katti/webkattis.py

```diff
@@ -162,6 +169,151 @@ def add_problem(problem_id: str, problem_conf: dict, kattis_config: configloader
     problem_conf[problem_id] = problem_rating
     configloader.problem_config_changed()
 
+def add_all_unfinished_problems(unsolved_problems_conf: dict, kattis_config: configloader.KattisConfig, verbose=False):
+    """Gets all unfinished problems from Kattis
+    
+    Parameters
+    ----------
+    kattis_config: configloader.KattisConfig
+        A KattisConfig object containing the user's kattis config
+    verbose: bool
+        A boolean flag to enable verbose mode
+
+    Warning
+    ----------
+    This function is slow and should only be used to initialize the unsolved problems config
+    """
+    global kattis_session
+    
+    login_response = login_with_password(kattis_config, verbose)
+
+    page_num = 1
+
+    # get the page number of pages of unsolved problems
+    url = f"{kattis_config.url}{_UNFINISHED_ENDING}"
+    r = kattis_session.get(url, headers=_HEADERS)
+
+    r_content = r.text
+    soup = BeautifulSoup(r_content, 'html.parser')
+    max_page_num = soup.find_all('div', class_='flex gap-2')[0].find_all('a')[2].text
+    max_page_num = int(max_page_num)
+
+    unsolved_problems_conf = {}
+
+    # for each page of problems get the problem id and rating and add to unsolved_problems
+    while page_num <= max_page_num:
+        print(f"Getting unsolved problems from page {page_num}/{max_page_num}...", end="\r", flush=True) if verbose else None
```

Comment:
Consider making this visible outside the verbose flag.  I'd rather users know what's happening for waits >3 seconds.